src/runtime_src/core/common/device.cpp

```diff
@@ -209,6 +210,23 @@ get_memidx_encoding(const uuid& xclbin_id) const
   return m_memidx_encoding;
 }
 
+device::memory_type
+device::
+get_memory_type(size_t memidx) const
+{
+  auto mem_topology = get_axlf_section<const ::mem_topology*>(ASK_GROUP_TOPOLOGY);
+  if (!mem_topology || mem_topology->m_count < static_cast<int32_t>(memidx))
```

Comment:
Minor item: Because of the static_cast and pointers, I missed the fact that we are basically doing `m_count < memidx` This oversight was due to obfuscation and I later went back to this line after reviewing line 221.

This code needs to be here, but i don't know what we can do to assist the next developer in seeing the intent. 

Again, just a minor issue I encountered while reviewing the code.